### ISSUE TITLE
[JS] [v1.3] Input labels should render as a <label for="X" /> the <input>

### ISSUE THREAD

**Issue by matthidinger** [microsoft/AdaptiveCards#4336]
# Platform

What platform is your issue or question related to? (Delete other platforms).

* JavaScript

# Details

I think there are opportunities to improve the HTML that is generated from the new `label` property on `Input` elements. If we take advantage of `<label for="X" />` we get a bunch of UX improvements outlined below.

## Payload

```json
{
    "type": "AdaptiveCard",
    "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
    "version": "1.3",
    "body": [
        {
            "type": "Input.Text",
            "placeholder": "Placeholder text",
            "label": "My label"
        }
    ]
}
```

## Generated HTML

```html
<div style="display: flex; flex-direction: column; box-sizing: border-box; flex: 0 0 auto;">
   <div class="ac-richTextBlock" id="__ac-af94ef97-a700-4531-9609-6b6de422c52e" style="text-align: left; box-sizing: border-box; flex: 0 0 auto; margin-bottom: 3px;">
       <span class="ac-textRun" style="font-family: Calibri, sans-serif; font-size: 14px; color: rgb(0, 0, 0); font-weight: 400; box-sizing: border-box; flex: 0 0 auto;">My label</span>
   </div>
   <div class="ac-input-container" style="display: flex;" data-children-count="1">
        <input class="ac-input ac-textInput" type="text" tabindex="0" placeholder="Placeholder text" aria-label="Placeholder text" aria-labelledby="__ac-af94ef97-a700-4531-9609-6b6de422c52e __ac-af94ef97-a700-4531-9609-6b6de422c52e" style="flex: 1 1 auto; min-width: 0px;">
     </div>
</div>
```

## Expectation

* Similar behavior as this: https://www.w3schools.com/tags/tryit.asp?filename=tryhtml5_input_type_text
* The output HTML semantically describes the text and inputs together 
* The mouse cursor over a label should be the standard cursor (not text selection)
* Clicking on a label will place focus in the input
